src/cms/views.py

```diff
@@ -106,7 +106,9 @@ def view_page(request, page_name):
                                      content_type=request.model_content_type,
                                      object_id=request.site_type.pk)
 
-    if request.journal:
+    if current_page.template:
+        template = current_page.template
```

Comment:
State in the database and actual theme code can diverge now, we should check page exists before trying to render (when not present, return 404 to user and log an error)

Reply:
Good call. I've added a check for the template path in the view, and tests to make sure the view logic works.